### DISCUSSION TITLE
Turtle texturing error

### DISCUSSION THREAD

**Discussion by MAGGen-hub** [cc-tweaked/CC-Tweaked#889] [Q&A] [answered]
I tried to make special upgrade for turtle, but when I try to use my modles:
    private static final ModelResourceLocation leftModel = new ModelResourceLocation( "ccbetterspeaker:turtle_speaker_upgrade_left", "inventory" );
    private static final ModelResourceLocation rightModel = new ModelResourceLocation( "ccbetterspeaker:turtle_speaker_upgrade_right", "inventory" );
I allways get black/purple missing texture block. What I do wrong? How can I fix this?

Code json:
turtle_speaker_upgrade_left.json
{
    "parent": "ccbetterspeaker:block/turtle_upgrade_base_left",
    "textures": {
        "texture": "ccbetterspeaker:block/turtle_speaker_advanced_face"
    }
}

turtle_upgrade_base_left.json
{
    "parent": "block/block",
    "textures": {
        "particle": "#texture"
    },
    "elements": [
        {
            "from": [ 0.5, 4.5, 3.5 ],
            "to": [ 2, 12.5, 11.5 ],
            "faces": {
                "down": { "uv": [ 2, 14, 14, 16 ], "texture": "#texture", "rotation": 270 },
                "up": { "uv": [ 2, 0, 14, 3 ], "texture": "#texture", "rotation": 90 },
                "north": { "uv": [ 0, 2, 3, 14 ], "texture": "#texture" },
                "south": { "uv": [ 13, 2, 16, 14 ], "texture": "#texture" },
                "west": { "uv": [ 2, 2, 14, 14 ], "texture": "#texture" }
            }
        }
    ]
}

**Comment by fatboychummy** [comment]
Please *please* format code with code blocks,
\`\`\`language-name
code
\`\`\`

## Java model code
```java
private static final ModelResourceLocation leftModel = new ModelResourceLocation( "ccbetterspeaker:turtle_speaker_upgrade_left", "inventory" );
private static final ModelResourceLocation rightModel = new ModelResourceLocation( "ccbetterspeaker:turtle_speaker_upgrade_right", "inventory" );
```

## `turtle_speaker_upgrade_left.json`
```json
{
	"parent": "ccbetterspeaker:block/turtle_upgrade_base_left",
	"textures": {
		"texture": "ccbetterspeaker:block/turtle_speaker_advanced_face"
	}
}
```

## `turtle_upgrade_base_left.json`
```json
{
	"parent": "block/block",
	"textures": {
		"particle": "#texture"
	},
	"elements": [
		{
			"from": [ 0.5, 4.5, 3.5 ],
			"to": [ 2, 12.5, 11.5 ],
			"faces": {
				"down": { "uv": [ 2, 14, 14, 16 ], "texture": "#texture", "rotation": 270 },
				"up": { "uv": [ 2, 0, 14, 3 ], "texture": "#texture", "rotation": 90 },
				"north": { "uv": [ 0, 2, 3, 14 ], "texture": "#texture" },
				"south": { "uv": [ 13, 2, 16, 14 ], "texture": "#texture" },
				"west": { "uv": [ 2, 2, 14, 14 ], "texture": "#texture" }
			}
		}
	]
}
```

**Reply by MAGGen-hub** [reply]
Sorry... You have any ideas?

**Reply by fatboychummy** [reply]
My only thought would be that you did `ccbetterspeaker:block/turtle_speaker_advanced_face`, possibly meant `_base` rather than `_face`? But not really sure'

I'm not great with modding yet though, just wanted to toss a formatted version of your post in here for other people to read.

**Reply by MAGGen-hub** [reply]
turtle_speaker_advanced_face - is thexture name... turtle_speaker_face is for normal speaker (found it it texture folder)

**Reply by MAGGen-hub** [reply]
May be I need to register the model, but I don't know HOW...

**Comment by SquidDev** [comment, marked as answer]
There's a `ModelRegistryEvent` which allows you to register "special models" - i.e. those not associated with a block or item. See what CC: Tweaked does:

https://github.com/SquidDev-CC/CC-Tweaked/blob/01ddb2b4e436d5931159c25c15eafdfe33f64de1/src/main/java/dan200/computercraft/client/ClientRegistry.java#L73-L81

**Reply by MAGGen-hub** [reply]
Thank you very much for your help. Now every part of my modification works correctly. I still need to add more lua functions to make the speaker_sound_layer_control system easier in use, but the main job is done. Here is the textures picture. Thank you for supporting me...
![2021-08-08_01 07 45](https://user-images.githubusercontent.com/56871670/128613875-1c664397-5bce-492b-9de3-c54fcdf0fca7.png)
